sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -285,6 +287,112 @@ case class ArrayTransform(
   override def prettyName: String = "transform"
 }
 
+/**
+ * Sorts elements in an array using a comparator function.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(expr, func) - Sorts the input array in ascending order. The elements of the " +
+    "input array must be orderable. Null elements will be placed at the end of the returned " +
+    "array. Since 3.0.0 also sorts and returns the array based on the given " +
+    "comparator function. The comparator will take two nullable arguments " +
+    "representing two nullable elements of the array." +
+    "It returns -1, 0, or 1 as the first nullable element is less than, equal to, or greater " +
+    "than the second nullable element. If the comparator function returns other " +
+    "values (including NULL), the query will fail and raise an error.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(5, 6, 1), (x, y) -> f(x, y));
+       [1,5,6]
+      > SELECT _FUNC_(array('bc', 'ab', 'dc'), (x, y) -> f(x, y));
+       ["dc", "bc", "ab"]
+      > SELECT _FUNC_(array('b', 'd', null, 'c', 'a'));
+ |     ["d", "c", "b", "a", null]
+  """,
+  since = "2.4.0")
+case class ArraySort(
+    argument: Expression,
+    function: Expression)
+  extends ArrayBasedSimpleHigherOrderFunction with ArraySortLike with CodegenFallback {
```

Comment:
Do we still need to extend `ArraySortLike`?

Reply:
Good catch. We may not need to extend `ArraySortLike` now.